cosmoz-page-location.html

```diff
@@ -1,238 +1,173 @@
 <link rel="import" href="../polymer/polymer.html">
 <link rel="import" href="../iron-location/iron-location.html">
 <link rel="import" href="../iron-location/iron-query-params.html">
+<!--
+The `cosmoz-page-location` element manages binding to and from the current URL.
 
+@demo demo/location.html
+-->
 <dom-module id="cosmoz-page-location">
 	<template>
-		<iron-location url-space-regex="^$"
-			path="{{appPath}}"
-			query="{{_urlQueryString}}"
-			hash="{{_urlHash}}">
+		<iron-location url-space-regex="^$" id="location"
+			path="{{ appPath }}"
+			query="{{ _appQueryString }}"
+			hash="{{ _appHashString }}">
 		</iron-location>
-		<iron-query-params
-			params-string="{{_urlQueryString}}"
-			params-object="{{appQueryParams}}">
-		</iron-query-params>
-		<iron-query-params
-			params-string="{{routeQueryString}}"
-			params-object="{{routeQueryParams}}">
-		</iron-query-params>
-		<iron-query-params
-			params-string="{{routeHash}}"
-			params-object="{{routeHashParams}}">
-		</iron-query-params>
-
-		<iron-query-params
-			params-string="{{_tempParamsString}}"
-			params-object="{{_tempParamsObject}}"></iron-query-params>
+
+		<iron-query-params id="params"
+		></iron-query-params>
+
 	</template>
 	<script>
 		(function () {
 			'use strict';
 
+			const defaultObject = function () {
+					return {};
+				},
+				assign = Object.assign;
```

Comment:
This hurts readability. Only used 3 times and it only saves 7 bytes each time, no perf difference.
If anything, make a "shallow copy object" wrapper rather since that's what it's used for every time.
But rather spell out Object.assign everywhere for now.